Adds example for process-file

agbilotia1998 opened the pull request:
Fixes #75 
Adds example for process-file in `process-file.md`

agbilotia1998 committed:
Adds example for process-file

agbilotia1998 commented:
@antho1404 please review.

agbilotia1998 commented:
Hi @antho1404 can you please review this PR and let me know if anything else needs to be done?

antho1404 commented:
Hello, sorry for the delay, thanks for the contribution.
That's good to have a complete example at the end but it would be better to have many small examples for each section with their different values.
- Trigger
- Task
- Filter
- Instance resolution

Example:

Trigger:
```yml
key: my-workflow
steps:
  - type: trigger
    instanceHash: XXX
    eventKey: eventX # listen to the event with the key `eventX`
    # or
    taskKey: taskX # listen to the result of the task with the key `taskX`
```

Every example should have the whole structure even if it is truncated.

Example:
```yml
key: my-workflow
steps:
  ...
  - type: filter
    conditions:
      dataX: xxx
  ... 
```

agbilotia1998 commented:
@antho1404 thanks for the review. Sure, I will add the examples for every section.

agbilotia1998 committed:
Adds example for all steps type in process-file

agbilotia1998 commented:
@antho1404 I have added examples to each of the section, I've added the different possibilities as I understood from the documentation. Please check if I missed something or any changes are required. Thanks :)

antho1404 commented on guide/process/process-file.md:
```diff
@@ -121,6 +133,22 @@ Reference the outputs of a previous step.
+      inputA: "Input1 to the task"
+      # or
+      stepKey: taskX
+      key: taskZ
```
This is a data of `inputA` so should be indented like
```
inputs: 
  inputA:
    stepKey: stepX
    key: taskX
```

This also needs to be updated in the complete example

antho1404 commented on guide/process/process-file.md:
```diff
@@ -156,10 +196,65 @@ All conditions should match to continue to the next step.
+key: erc20-notification
+steps:
+  - type: trigger
+    instance:
```
One example with the `instanceHash` would be great too like that we have the 3 different versions

agbilotia1998 committed:
Corrects reference parameters to input and adds instanceHash example

agbilotia1998 commented:
@antho1404  I've made the changes, please review.

antho1404 reviewed (approved):
Thanks @agbilotia1998, that's great
